### ISSUE TITLE
[DEFECT]student profile pic is blank (not empty) at the beginning, then will appear later

### ISSUE THREAD

**Issue by lance-huang-nucloudglobal** [nucloudglobal/lms-ext#38]
Step:
1. LMS
2.Check the course which has integrated user
3. people
4. view student profile

Expected result
user profile pic shall be integrated into LMS right after the user is integrated
Manage user account -> View student LMS profile

SS
![image](https://cloud.githubusercontent.com/assets/8621588/11140114/d08b40f4-8a11-11e5-97bc-2e723a19f703.png)

**Comment by jbaltazar** [comment]
I think the reason why the avatar in LMS is empty after integration is the cron time.
[CronExport.run_them_all](https://github.com/nucloudglobal/lms/blob/dev/app/models/cron_export.rb#L136-L140) is same and exact time of run with [CronExport.sync_avatar](https://github.com/nucloudglobal/lms/blob/dev/app/models/cron_export.rb#L85-L134)

[See reference](https://github.com/nucloudglobal/lms/blob/dev/config/schedule.rb#L26-L29)

you will see here that two function is running every 2 hours. but it should run first the [CronExport.run_them_all](https://github.com/nucloudglobal/lms/blob/dev/app/models/cron_export.rb#L136-L140) due of this is the function for integrating user then after the integration the [CronExport.sync_avatar](https://github.com/nucloudglobal/lms/blob/dev/app/models/cron_export.rb#L85-L134) will run next to check and update the avatar.


To summarize and make it short, the two function is running on parallel so upon integrating the user from ECW to LMS, it's the same time that other function is checking on ECW avatar if any. but due of the user is not yet done for integration. The avatar will be left as empty and it will not going to update and it will only be updated after hours.

[Pull Request Info and Dependencies](https://github.com/nucloudglobal/lms/pull/10)

**Comment by lance-huang-nucloudglobal** [comment]
@jbaltazar 
Thanks for working in this time.
just to remind, on UAT, crons shall run per 30 mins(I assume all integration cron show run in this frequency?), not per 2 hours like on TEST or per 24 hours according to FS.

CC @apetitnucloud

**Comment by lance-huang-nucloudglobal** [comment]
@apetitnucloud 
removed Code Review since code has been reviewed

CC @jbaltazar

**Comment by nucloud-agutierrez** [comment]
QA20160113: QA tested. Fixed.

*see attached screenshot*
![screen shot 2016-01-13 at 9 09 16 am](https://cloud.githubusercontent.com/assets/8049330/12281930/33c37ec8-b9d5-11e5-8f28-b357dc166bf0.png)

**Comment by lance-huang-nucloudglobal** [comment]
@nucloud-agutierrez 
I do not see the steps of testing. so I would like to confirm here with you, did you follow the steps?
the important thing is 

Expected result
user profile pic shall be integrated into LMS **right after** the user is integrated
Manage user account -> View student LMS profile

That means the issue shall be proof to fixed if we 
1. Setup a student who has profile pic in ECW/CRM shall be integrated (this student has not been integrated into LMS before)
2. Wait the integration to happen
3. Check the LMS to see if the student become the user+people
4. Check if the LMS user has right profile pic

**Comment by nucloud-agutierrez** [comment]
@lance-huang-nucloudglobal , I don't know what happened, why my profile pic is not showing in your test but it shows on mine. And yes, those were the steps I used.

However, upon doing it again to replicate the issue you encountered, that the profile pic is not showing, I encounter this issue, error page upon registering in ecwsqa.

*see attached screenshot*

![screen shot 2016-01-14 at 9 48 44 am](https://cloud.githubusercontent.com/assets/8049330/12313623/2753a364-baa4-11e5-97d5-e629ab16cf5d.png)

cc: @apetitnucloud

**Comment by lance-huang-nucloudglobal** [comment]
@nucloud-agutierrez 
you may want to confirm on skype with Aurelien the frequency of all integration cronjob.(It has been too long since we confirm it last time, sth may change)
and consider is it possible that 2nd round of the cronjob overwrite sth could be the reason I see empty profile pic.

**Comment by nucloud-agutierrez** [comment]
QA20150121: QA tested. Fixed in Test Environment.

1. Setup a student who has profile pic in ECW/CRM shall be integrated (this student has not been integrated into LMS before)
2. Wait the integration to happen
3. Check the LMS to see if the student become the user+people
4. Check if the LMS user has right profile pic

*see attached test evidence*

![screen shot 2016-01-21 at 11 51 39 am](https://cloud.githubusercontent.com/assets/8049330/12471112/50deff6c-c035-11e5-8a14-67c9da4f8447.png)

![screen shot 2016-01-21 at 11 51 51 am](https://cloud.githubusercontent.com/assets/8049330/12471114/58c18538-c035-11e5-8ea5-c904c9edc3e6.png)
